### PR TITLE
fix the Golang code for median in quick_sort.md

### PR DESCRIPTION
**Pull request by dshlstarr**
Fix the Golang code for median

If this PR is related to coding or code translation, please fill out the checklist.

- [x] I've tested the code and ensured the outputs are the same as the outputs of reference codes.
- [x] I've checked the codes (formatting, comments, indentation, file header, etc) carefully.
- [x] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).

### CODE REVIEW

File: docs/chapter_sorting/quick_sort.md [hunk 1]
```diff
@@ -483,7 +483,7 @@ comments: true
     func medianThree(nums []int, left, mid, right int) int {
         if (nums[left] > nums[mid]) != (nums[left] > nums[right]) {
             return left
-        } else if (nums[mid] < nums[left]) != (nums[mid] > nums[right]) {
+        } else if (nums[mid] > nums[left]) != (nums[mid] > nums[right]) {
```
Comment (Reanon) [inline docs/chapter_sorting/quick_sort.md:486]: There is no error, you can test the code.

**Review by krahets — APPROVED** [review]
Thanks for the fix!

File: docs/chapter_sorting/quick_sort.md (hunk 1 above)
Comment (Reanon) [inline docs/chapter_sorting/quick_sort.md:486]: It seems that this problem is also found in other code, you can modify the codes meanwhile.

Comment (krahets) [reply docs/chapter_sorting/quick_sort.md:486]: > It seems that this problem is also found in other code, you can modify the codes meanwhile.

No need to fix other codes.

**Pull request merged**